activesupport/lib/active_support/callbacks.rb

```diff
@@ -79,12 +79,21 @@ module Callbacks
     #   run_callbacks :save do
     #     save
     #   end
+
+    # Store a frozen copy of a callback string so it doesn't need to be
+    # allocated and de-allocated for every call
+    @@_callbacks_store = {}
+
     def run_callbacks(kind, &block)
-      send "_run_#{kind}_callbacks", &block
+      send callbacks_store(kind), &block
     end
 
     private
 
+    def callbacks_store(kind)
+      @@_callbacks_store[kind] ? @@_callbacks_store[kind] : @@_callbacks_store[kind]  = "_run_#{kind}_callbacks".freeze
```

Comment:
Would this work?

`@@_callbacks_store[kind] ||= "_run_#{kind}_callbacks".freeze`